src/backend/api/Fusion.Resources.Domain/Queries/GetDepartments.cs

```diff
@@ -35,24 +49,102 @@ public GetDepartments StartsWith(string department)
             this.departmentFilter = department;
             return this;
         }
+
+        public GetDepartments ById(string departmentId)
+        {
+            this.departmentId = departmentId;
+            return this;
+        }
+
         public GetDepartments InSector(string sector)
         {
             this.sector = sector;
             return this;
         }
 
+        public GetDepartments ExpandResourceOwners()
+        {
+            shouldExpandResourceOwners = true;
+            return this;
+        }
+
+        public GetDepartments WhereResourceOwnerMatches(string search)
+        {
+            resourceOwnerSearch = search;
+            return this;
+        }
+
         public class Handler : IRequestHandler<GetDepartments, IEnumerable<QueryDepartment>>
         {
             private readonly ResourcesDbContext db;
+            private readonly IHttpClientFactory httpClientFactory;
+            private readonly IFusionProfileResolver profileResolver;
 
-            public Handler(ResourcesDbContext db)
-            {
+            public Handler(ResourcesDbContext db, IHttpClientFactory httpClientFactory,
+                IFusionProfileResolver profileResolver)
+            { 
                 this.db = db;
+                this.httpClientFactory = httpClientFactory;
+                this.profileResolver = profileResolver;
             }
 
             public async Task<IEnumerable<QueryDepartment>> Handle(GetDepartments request, CancellationToken cancellationToken)
             {
-                return await request.Execute(db.Departments).ToListAsync(cancellationToken);
+                var departments = await request.Execute(db.Departments).ToListAsync(cancellationToken);
+
+                if(request.shouldExpandResourceOwners)
+                {
+                    return await ExpandResourceOwners(departments, request.resourceOwnerSearch, cancellationToken);
+                }
+
+                return departments;
+            }
+
+            private async Task<List<QueryDepartment>> ExpandResourceOwners(List<QueryDepartment> departments, string? filter, CancellationToken cancellationToken)
+            {
+                var result = new List<QueryDepartment>();
+                var managedDepartments = departments.ToDictionary(dpt => dpt.DepartmentId);
+
+                var client = httpClientFactory.CreateClient("lineorg");
+
+                var uri = "/lineorg/persons?$filter=isresourceowner eq true";
```

Comment:
There is no upper limit on the page size in the lineorg api. You could just set the page size to 2k. This is a plain lookup in the database for that api.